pkg/collector/pods_containerlogs_collector.go

```diff
@@ -0,0 +1,148 @@
+package collector
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	restclient "k8s.io/client-go/rest"
+)
+
+// PODSContainerLogsCollector defines a Pods Container Logs Collector struct
+type PODSContainerLogsCollector struct {
+	kubeconfig *restclient.Config
+	data       map[string]string
+}
+
+// NewPODSContainerLogs is a constructor
+func NewPODSContainerLogs(config *restclient.Config) *PODSContainerLogsCollector {
+	return &PODSContainerLogsCollector{
+		data:       make(map[string]string),
+		kubeconfig: config,
+	}
+}
+
+func (collector *PODSContainerLogsCollector) GetName() string {
+	return "podscontainerlogs"
+}
+
+// Collect implements the interface method
+func (collector *PODSContainerLogsCollector) Collect() error {
+	containerNamespaces := strings.Fields(os.Getenv("DIAGNOSTIC_CONTAINERLOGS_LIST"))
+
+	// Creates the clientset
+	clientset, err := kubernetes.NewForConfig(collector.kubeconfig)
+	if err != nil {
+		return fmt.Errorf("error in getting access to K8S")
```

Comment:
this line should pop a warning in vscode because you didn't pass the error

Reply:
It interesting din't but got this covered in next commit.